### ISSUE TITLE
lineSlice bug

### ISSUE THREAD

**Issue by mourner** [Turfjs/turf#351]
Test case:

``` js
var line = [[-96.914526,32.836763],[-96.914155,32.836758],[-96.9104,32.836758]];
var start = [-96.9132882334045,32.836758];
var stop = [-96.91163781431621,32.836758];

var turfSlice = turf.lineSlice(turf.point(start), turf.point(stop), turf.linestring(line));
```

Expected:
![image](https://cloud.githubusercontent.com/assets/25395/14752108/fbd6d5c8-08d5-11e6-93b9-2184d3d06eb9.png)

Actual:
![image](https://cloud.githubusercontent.com/assets/25395/14752083/d3fbfad8-08d5-11e6-8637-7e803516802f.png)

cc @morganherlocker @tcql

Discovered this while implementing `lineSlice` in `cheap-ruler`. This is the only failing case in my fixtures (from a Tiger tile).

**Comment by psnider** [comment]
Runing the following from the node REPL:
```
turf = require('turf')
let coords = [[0,0],[1,0]]
let line = turf.linestring(coords)
let start = turf.point([0.5,0])
let stop  = turf.point([0.7,0])
let slice = turf.lineSlice(start, stop, line)
slice.geometry.coordinates
```
displays:
> [ [ 0, 0 ], [ 1, 0 ] ]

while I expected:
> [ [ 0.5, 0 ], [ 0.7, 0 ] ]

Is this is an error, or expected behavior?

**Comment by psnider** [comment]
And changing the above example slightly yields a result with a duplicated point:
```
turf = require('turf')
let coords = [[0,0],[1,0],[2,0]]
let line = turf.linestring(coords)
let start = turf.point([1.5,0])
let stop  = turf.point([1.7,0])
let slice = turf.lineSlice(start, stop, line)
slice.geometry.coordinates
```
displays:
> [ [ 1, 0 ], [ 1, 0 ], [ 2, 0 ] ]

which is certainly incorrect.

**Comment by psnider** [comment]
Looking at the code, https://github.com/Turfjs/turf/blob/master/packages/turf-line-slice/index.js#149, 
it seems that the boundary conditions are incorrect.

These values from the above example fail:
```
line1StartX=1.5
line1StartY=0
line1EndX=1.5000000000000018
line1EndY=-14.468631190172305

line2StartX=1
line2StartY=0
line2EndX=2
line2EndY=0

result = lineIntersects(line1StartX, line1StartY, line1EndX, line1EndY, line2StartX, line2StartY, line2EndX, line2EndY)
```

**Comment by psnider** [comment]
Please cite the algorithm you use in the code.

**Comment by psnider** [comment]
I'm guessing that the algorithm is supposed to be:
https://en.wikipedia.org/wiki/Line%E2%80%93line_intersection#Given_two_points_on_each_line

Then it seems that these tests are incorrect:
https://github.com/Turfjs/turf/blob/master/packages/turf-line-slice/index.js#L178-184
and must be changed from open intervals to closed:
```
if (a >= 0 && a <= 1) {
    result.onLine1 = true;
  }
  // if line2 is a segment and line1 is infinite, they intersect if:
  if (b >= 0 && b <= 1) {
    result.onLine2 = true;
  }
```

**Comment by psnider** [comment]
I looked at making a pull requests, but I can't find any automated tests for this package.
Instead the tests I found write GeoJSON files, which I assume someone looks at by hand?

I won't create a pull request without automated tests.
Please advise.

**Comment by mourner** [comment]
@psnider the automated tests are here: https://github.com/Turfjs/turf/blob/master/packages/turf-line-slice/test.js
We'll be happy to see a pull request with a bugfix from you!

**Comment by dpmcmlxxvi** [comment]
Appears to have been fixed by 636b34b. Here is a [jsfiddle](https://jsfiddle.net/0hoLcqrq/4/). Recommend closing.

**Comment by morganherlocker** [comment]
Thanks for flagging @dpmcmlxxvi!
